CleverTapSDK/InApps/CTInAppEvaluationManager.m

```diff
@@ -116,8 +151,18 @@ - (void)evaluateOnAppLaunchedServerSide:(NSArray *)appLaunchedNotifs {
     }
 }
 
-- (void)evaluateClientSide:(CTEventAdapter *)event {
-    NSMutableArray *eligibleInApps = [self evaluate:event withInApps:self.inAppStore.clientSideInApps];
+- (void)evaluateClientSide:(NSArray<CTEventAdapter *> *)events {
+    NSMutableArray<NSDictionary *> *eligibleInApps = [NSMutableArray array];    
+    for (CTEventAdapter *event in events) {
+        id oldValue = [event.eventProperties objectForKey:CLTAP_KEY_OLD_VALUE];
+        id newValue = [event.eventProperties objectForKey:CLTAP_KEY_NEW_VALUE];
+        if (event.profileAttrName != nil) {
+            if (newValue == oldValue) {
+                continue;
+            }
+            [eligibleInApps addObjectsFromArray:[self evaluate:event withInApps:self.inAppStore.clientSideInApps]];
```

Comment:
The evaluation needs to be outside the if event.profileAttrName so non-profile events are also evaluated